File: src/missionpkg/src/video.py
#! /usr/bin/env python

import roslib
import numpy as np
import rospy
import cv2
import sys, time
from sensor_msgs.msg import Image
from cv_bridge import CvBridge, CvBridgeError

v1 = np.zeros((480, 640), np.uint8)
v2 = np.zeros((480, 640), np.uint8)

class ellipse_detection:

	def __init__(self):
        	cv2.namedWindow('mywindow')
        	cv2.namedWindow('mywindow1')
        	self.bridge = CvBridge()
        	self.image_sub = rospy.Subscriber("usb_cam/image_raw",Image,self.callback)
		self.image_sub1 = rospy.Subscriber("usb_cam1/image_raw",Image,self.callback1)

	def callback(self,data):
        	try:
        		global v1
           		v1 = self.bridge.imgmsg_to_cv2(data, "bgr8")
        	except CvBridgeError as e:
           		print(e)

    	def callback1(self,data):
        	try:
	    		global v2 
	    		v2 = self.bridge.imgmsg_to_cv2(data, "bgr8")
           		self.showImage()
        	except CvBridgeError as e:
            		print(e)


	def showImage(self):
	        imgray2 = cv2.cvtColor(v1, cv2.COLOR_BGR2GRAY)
	        imgray1 = cv2.cvtColor(v2, cv2.COLOR_BGR2GRAY)
	        cv2.imshow('mywindow1', imgray2)
	        cv2.imshow('mywindow', imgray1)
        	cv2.waitKey(1)   

def main(args):
	ed = ellipse_detection()
	#showImage()
	rospy.init_node('qwerty', anonymous=True)
	try:
		rospy.spin()
	except KeyboardInterrupt:
        	print("Shutting down ellipse_detection")
        
        cv2.destroyAllWindows()

if __name__ == '__main__':
	main(sys.argv)

File: src/missionpkg/src/video.cpp
#include <ros/ros.h>
#include <image_transport/image_transport.h>
#include <cv_bridge/cv_bridge.h>
#include <sensor_msgs/image_encodings.h>
#include <opencv2/imgproc/imgproc.hpp>
#include <opencv2/highgui/highgui.hpp>

static const std::string OPENCV_WINDOW = "Image window";
static const std::string OPENCV_WINDOW1 = "Image window1";

class ImageConverter
{
  ros::NodeHandle nh_;
  image_transport::ImageTransport it_;
  image_transport::Subscriber image_sub_;
  image_transport::Subscriber image_sub1_;
//  image_transport::Publisher image_pub_;
//  image_transport::Publisher image_pub1_;

public:
  ImageConverter()
    : it_(nh_)
  {
    // Subscribe to input video feed and publish output video feed
    image_sub_ = it_.subscribe("/usb_cam/image_raw", 1,
      &ImageConverter::imageCb, this);
    image_sub1_ = it_.subscribe("/usb_cam1/image_raw", 1,
      &ImageConverter::imageCb1, this);

    cv::namedWindow(OPENCV_WINDOW);
    cv::namedWindow(OPENCV_WINDOW1);
  }

  ~ImageConverter()
  {
    cv::destroyWindow(OPENCV_WINDOW);
    cv::destroyWindow(OPENCV_WINDOW1);
  }

  void imageCb1(const sensor_msgs::ImageConstPtr& msg)
  {
    cv_bridge::CvImagePtr cv_ptr1;
    try
    {
      cv_ptr1 = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
    }
    catch (cv_bridge::Exception& e)
    {
      ROS_ERROR("cv_bridge exception: %s", e.what());
      return;
    }

    // Update GUI Window
    cv::imshow(OPENCV_WINDOW1, cv_ptr1->image);
    cv::waitKey(3);

    // Output modified video stream
//    image_pub1_.publish(cv_ptr1->toImageMsg());
  }

  void imageCb(const sensor_msgs::ImageConstPtr& msg)
  {
    cv_bridge::CvImagePtr cv_ptr;
    try
    {
      cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
    }
    catch (cv_bridge::Exception& e)
    {
      ROS_ERROR("cv_bridge exception: %s", e.what());
      return;
    }

    // Update GUI Window
    cv::imshow(OPENCV_WINDOW, cv_ptr->image);
    cv::waitKey(3);

    // Output modified video stream
//    image_pub_.publish(cv_ptr->toImageMsg());
  }
};

int main(int argc, char** argv)
{
  ros::init(argc, argv, "image_converter");
  ImageConverter ic;
  ros::spin();
  return 0;
}
